tests/src/Interop/PInvoke/DateTime/DateTimeTest.cs

```diff
@@ -7,172 +7,46 @@
-    private static extern Del_Marshal_InOut_cdecl GetDel_Marshal_InOut_cdecl();
-
-    [UnmanagedFunctionPointer(CallingConvention.StdCall)]
-    delegate bool Del_Marshal_Out_stdcall([Out][MarshalAs(UnmanagedType.Struct)] out DateTime t);
-
-    [DllImport("NativeDateTime.dll", CallingConvention = CallingConvention.StdCall)]
-    private static extern Del_Marshal_Out_stdcall GetDel_Marshal_Out_stdcall();
-
-    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-    delegate bool Del_MarshalSeqStruct_InOut_cdecl([In, Out][MarshalAs(UnmanagedType.Struct)] ref Stru_Seq_DateAsStructAsFld t);
-
-    [DllImport("NativeDateTime.dll", CallingConvention = CallingConvention.StdCall)]
-    private static extern Del_MarshalSeqStruct_InOut_cdecl GetDel_Del_MarshalSeqStruct_InOut_cdecl();
-
-    [UnmanagedFunctionPointer(CallingConvention.Cdecl)]
-    delegate bool Del_MarshalExpStruct_InOut_cdecl([In, Out][MarshalAs(UnmanagedType.Struct)] ref Stru_Exp_DateAsStructAsFld t);
-
-    [DllImport("NativeDateTime.dll", CallingConvention = CallingConvention.StdCall)]
-    private static extern Del_MarshalExpStruct_InOut_cdecl GetDel_Del_MarshalExpStruct_InOut_cdecl();
-
-    #endregion
-
-    #region ReversePInvoke
-
-    [DllImport("NativeDateTime.dll", CallingConvention = CallingConvention.Cdecl)]
-    private static extern bool RevP_Marshal_InOut_cdecl(Del_Marshal_InOut_cdecl d);
-
-    public static bool RevPMethod_Marshal_InOut_cdecl(ref DateTime d)
-    {
-        ExpectedRetdate = new DateTime(1947, 8, 15);
-        Assert.AreEqual(ExpectedRetdate, d, "RevPMethod_Marshal_InOut_cdecl : Date didn't match to expected date");
-        d = d.AddDays(-1);
-        return true;
-    }
-
-    #endregion
-
     static int Main(string[] args)
     {
-        try{
-            ExpectedRetdate = new DateTime(1947, 8, 15);
-
-            #region Pinvoke
-            DateTime Date1 = new DateTime(2008, 7, 4);
-            Assert.IsTrue(Marshal_In_stdcall(Date1), "Marshal_In_stdcall : Returned false");
-            Assert.IsTrue(Marshal_InOut_cdecl(ref Date1), "Marshal_InOut_cdecl : Returned false");
-            Assert.AreEqual(ExpectedRetdate, Date1, "Marshal_InOut_cdecl : Returned date is wrong");
-            
-            DateTime Date2;
-            Assert.IsTrue(Marshal_Out_stdcall(out Date2), "Marshal_In_stdcall : Returned false");
-            Assert.AreEqual(ExpectedRetdate, Date2, "Marshal_InOut_cdecl : Returned date is wrong");
-
-            Stru_Seq_DateAsStructAsFld StDate1;
-            StDate1.dt = new DateTime(2008, 7, 4);
-            StDate1.iInt = 100;
-            StDate1.bstr = "Managed";
-            Assert.IsTrue(MarshalSeqStruct_InOut_cdecl(ref StDate1), "MarshalSeqStruct_InOut_cdecl : Native side check failed");
-            Assert.AreEqual(ExpectedRetdate, StDate1.dt, "MarshalSeqStruct_InOut_cdecl : Returned date is wrong");
+        try
+        {
+            DateTime currentDate = new DateTime(2019, 5, 2);
```

Comment:
Instead of hardcoding this, it might be beneficial to instead use `DateTime.Now`. The benefit of this is that it is predictable _and_ we will properly test all cases through time (i.e. leap year, month nonsense, etc). Would any of those things even apply?

Reply:
You could also start from `DateTime.Now` and in a loop go through 366 days to test all the scenarios for a year in the future. Unsure how long the test would take though.

Reply:
I don't think that date-time nonsense would really apply here. Under the hood the conversion is just converting to/from the number of "ticks". The actual date-time nonsense should be tested in our tests for the DateTime structure.